uplink/storage/segments/store_test.go

```diff
@@ -4,212 +4,18 @@
 package segments_test
 
 import (
-	"bytes"
-	"context"
 	"fmt"
-	"io/ioutil"
-	"strconv"
 	"testing"
-	time "time"
 
 	"github.com/stretchr/testify/assert"
-	"github.com/stretchr/testify/require"
-	"github.com/vivint/infectious"
 
-	"storj.io/storj/internal/memory"
-	"storj.io/storj/internal/testcontext"
-	"storj.io/storj/internal/testplanet"
-	"storj.io/storj/internal/testrand"
-	"storj.io/storj/pkg/macaroon"
-	"storj.io/storj/pkg/pb"
-	storj "storj.io/storj/pkg/storj"
-	"storj.io/storj/satellite/console"
-	"storj.io/storj/storage"
-	"storj.io/storj/uplink/ecclient"
-	"storj.io/storj/uplink/eestream"
-	"storj.io/storj/uplink/storage/meta"
+	"storj.io/storj/pkg/storj"
 	"storj.io/storj/uplink/storage/segments"
 )
 
-func TestSegmentStoreMeta(t *testing.T) {
-	for i, tt := range []struct {
-		path       string
-		data       []byte
-		metadata   []byte
-		expiration time.Time
-		err        string
-	}{
-		{"l/path/1/2/3", []byte("content"), []byte("metadata"), time.Now().UTC().Add(time.Hour * 12), ""},
-		{"l/not-exists-path/1/2/3", []byte{}, []byte{}, time.Now(), "key not found"},
-		{"", []byte{}, []byte{}, time.Now(), "invalid segment component"},
-	} {
-		test := tt
-		t.Run("#"+strconv.Itoa(i), func(t *testing.T) {
-			runTest(t, func(t *testing.T, ctx *testcontext.Context, planet *testplanet.Planet, segmentStore segments.Store) {
-				expectedSize := int64(len(test.data))
-				reader := bytes.NewReader(test.data)
-
-				beforeModified := time.Now()
-				if test.err == "" {
-					meta, err := segmentStore.Put(ctx, reader, test.expiration, func() (storj.Path, []byte, error) {
-						return test.path, test.metadata, nil
-					})
-					require.NoError(t, err)
-					assert.Equal(t, expectedSize, meta.Size)
-					assert.Equal(t, test.metadata, meta.Data)
-					assert.True(t, test.expiration.Equal(meta.Expiration))
-					assert.True(t, meta.Modified.After(beforeModified))
-				}
-
-				meta, err := segmentStore.Meta(ctx, test.path)
-				if test.err == "" {
-					require.NoError(t, err)
-					assert.Equal(t, expectedSize, meta.Size)
-					assert.Equal(t, test.metadata, meta.Data)
-					assert.True(t, test.expiration.Equal(meta.Expiration))
-					assert.True(t, meta.Modified.After(beforeModified))
-				} else {
-					require.Contains(t, err.Error(), test.err)
-				}
-			})
-		})
-	}
-}
-
-func TestSegmentStorePutGet(t *testing.T) {
```

Comment:
Those tests were moved to `streams/store_test.go` and adjusted to new metainfo API. In general with new metainfo API it make no sense to test put/get on segment level because now we need also to create object to be able to create segment.